stats/view/aggregation_data.go

```diff
@@ -139,26 +159,61 @@ func (a *DistributionData) addSample(f float64) {
 	a.SumOfSquaredDev = a.SumOfSquaredDev + (f-oldMean)*(f-a.Mean)
 }
 
-func (a *DistributionData) incrementBucketCount(f float64) {
-	if len(a.bounds) == 0 {
-		a.CountPerBucket[0]++
+func (a *DistributionData) addToBucket(val float64, ex *exemplar.Exemplar) {
+	var (
+		count  *int64
+		bucket *Bucket
+	)
+
+	for i, b := range a.bounds {
+		if val < b {
+			count = &a.CountPerBucket[i]
+			bucket = &a.Buckets[i]
+			break
+		}
+	}
+
+	if count == nil {
+		count = &a.CountPerBucket[len(a.bounds)]
+		bucket = &a.Buckets[len(a.bounds)]
+	}
+
+	bucket.Count++
+	*count = bucket.Count
+	bucket.maybeRetainExemplar(val, ex)
+}
+
+func (b *Bucket) maybeRetainExemplar(val float64, e *exemplar.Exemplar) {
+	if b.Exemplar == nil {
+		b.Exemplar = &ExemplarData{
+			Value:       val,
+			Timestamp:   e.Timestamp,
+			Attachments: e.Attachments,
+		}
 		return
 	}
 
-	for i, b := range a.bounds {
-		if f < b {
-			a.CountPerBucket[i]++
-			return
+	// Heuristic to pick the "better" exemplar: first keep the one with a
+	// sampled trace attachment, if neither have a trace attachment, pick the
+	// one with more attachments.
+	_, haveTraceID := e.Attachments["TraceID"]
+	if haveTraceID || len(e.Attachments) >= len(b.Exemplar.Attachments) {
```

Comment:
I believe we meant to use `>` instead of `>=` otherwise it'll always create a new Exempler if `!haveTraceID && len(e.Attachments) == len(b.Exemplar.Attachments)`

Reply:
Yeah, the idea is that we replace the existing exemplar with the newer one if they are equally "good" under the assumption that newer is better.